### PR TITLE
Trainee form

### LINKED ISSUE #1: Create form to make trainee card
As a trainee i want to be able to add 

- my name, 
- github link, 
- linkedin link,  
- my CV link, 
- portfolio link and 
- a short introduction about myself 

to the form so that my card is visible on the main page

### PR DESCRIPTION
**Pull request by AndrianaOS**
## Naming Rules

Name your PR like this: #1 - Create form to make trainee form - Andriana

## Description

Created a trainee registration form to be submitted by trainee to go into the database and information to be displayed on the graduate page.
Form areas include graduates':
- Name
- GitHub Link
- Portfolio Link
- LinkedIn Link
- About themselves
- their skills and 
- their role

## Related to

Make sure you include the issue number with a hash sign # so Github can link this PR to the right issue, like this:

Fixes #1

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have carefully reviewed my own code
- [ ] I have commented my code
- [ ] I have updated any documentation

### CODE REVIEW

File: frontend/src/components/TraineeForm/TraineeForm.jsx [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+      trainee_name: formDetails.trainee_name,
+      github_link: formDetails.github_link,
+      portfolio_link: formDetails.portfolio_link,
+      linkedIn_link: formDetails.linkedIn_link,
+      role: formDetails.role,
+      about_me: formDetails.about_me,
+      skills: formDetails.skills,
+    };
+    try {
+      await postFormData(data);
```
Comment (PeteLindsell) [inline frontend/src/components/TraineeForm/TraineeForm.jsx:61]: I dont think there is any need to create a new object here
```suggestion
    try {
      await postFormData(formDetails);
```

File: frontend/src/components/TraineeForm/TraineeForm.jsx [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+      setFormDetails(formInitialState);
```
Comment (PeteLindsell) [inline frontend/src/components/TraineeForm/TraineeForm.jsx:62]: There is no need to call this function here and in postFormData

File: frontend/src/components/TraineeForm/TraineeForm.jsx [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+          name="linkedIn_link"
+          className="data_input"
+          value={formDetails.linkedIn_link}
+          onChange={handleInputChange}
+          required
+        />
+      </div>
+
+      <fieldset className="role">
+        <legend>Role:</legend>
```
Comment (PeteLindsell) [inline frontend/src/components/TraineeForm/TraineeForm.jsx:123]: Good use of fieldset and legend here 👍

**Review by PeteLindsell — COMMENTED** [review]
Lots of good stuff here! I have added a few comments to look at. I will leave it to others in the team to approve.

**Review by Zobeir-Rigi — APPROVED** [review]

**Pull request merged**